File: lab_5_Curve_Fitting_Straight_line.cpp
// Curve Fitting of a straight line using Least Squares Method
#include <iostream>
#include <iomanip>
using namespace std;

int main()
{
    int n;
    cout << "Enter number of data points: ";
    cin >> n;

    double x[n], y[n];
    double sumx = 0, sumy = 0, sumxy = 0, sumx2 = 0;

    cout << "Enter x and y values:\n";
    for(int i = 0; i < n; i++)
    {
        cin >> x[i] >> y[i];

        sumx += x[i];
        sumy += y[i];
        sumxy += x[i] * y[i];
        sumx2 += x[i] * x[i];
    }

    double b = (n * sumxy - sumx * sumy) / (n * sumx2 - sumx * sumx);
    double a = (sumy - b * sumx) / n;

    cout << fixed << setprecision(6);

    cout << "\nFitted line equation:\n";
    cout << "y = " << a << " + " << b << "x" << endl;

    return 0;
}

File: lab_5_Curve_Fitting_Straight_line.py
n = int(input("Enter number of data points: "))

x = []
y = []

sumx = 0
sumy = 0
sumxy = 0
sumx2 = 0

print("Enter x and y values:")

for i in range(n):
    xi = float(input("x: "))
    yi = float(input("y: "))

    x.append(xi)
    y.append(yi)

    sumx += xi
    sumy += yi
    sumxy += xi * yi
    sumx2 += xi * xi

b = (n * sumxy - sumx * sumy) / (n * sumx2 - sumx * sumx)
a = (sumy - b * sumx) / n

print("\nFitted line equation:")
print("y = {:.6f} + {:.6f}x".format(a, b))